async-nats/src/lib.rs

```diff
@@ -519,13 +522,24 @@ impl ConnectionHandler {
                 let mut made_progress = true;
                 loop {
                     while !self.handler.connection.is_write_buf_full() {
-                        match self.receiver.poll_recv(cx) {
+                        debug_assert!(self.recv_buf.is_empty());
```

Comment:
Do you think it's worth keeping this one for eventual catching of bugs later?

Reply:
Yeah I'd keep it. It should come in handy when adding websocket support or other changes.